pkg/api/testing/unstructured_test.go

```diff
@@ -134,6 +142,208 @@ func TestRoundTrip(t *testing.T) {
 	}
 }
 
+// TestRoundtripToUnstructured verifies the roundtrip faithfulness of all external types from native
+// to unstructured and back using both the JSON and CBOR serializers. The intermediate unstructured
+// objects produced by both encodings must be identical and be themselves roundtrippable to JSON and
+// CBOR.
+func TestRoundtripToUnstructured(t *testing.T) {
+	// These are GVKs that whose CBOR roundtrippability is blocked by a known issue that must be
+	// resolved as a prerequisite for alpha.
+	knownFailureReasons := map[string][]schema.GroupVersionKind{
+		// Since JSON cannot directly represent arbitrary byte sequences, a byte slice
+		// encodes to a JSON string containing the base64 encoding of the slice
+		// contents. Decoding a JSON string into a byte slice assumes (and requires) that
+		// the JSON string contain base64-encoded data. The CBOR serializer must be
+		// compatible with this behavior.
+		"byte slices should be represented in unstructured as base64-encoded strings": {
+			{Version: "v1", Kind: "Secret"},
+			{Version: "v1", Kind: "SecretList"},
+			{Version: "v1", Kind: "RangeAllocation"},
+			{Version: "v1", Kind: "ConfigMap"},
+			{Version: "v1", Kind: "ConfigMapList"},
+			{Group: "admissionregistration.k8s.io", Version: "v1beta1", Kind: "MutatingWebhookConfiguration"},
```

Comment:
Should APIService be here, too? https://github.com/kubernetes/kubernetes/blob/68091805a53c6fdd42553f8ccdbf932ab263fbdd/staging/src/k8s.io/kube-aggregator/pkg/apis/apiregistration/v1/types.go#L69
Perhaps it's not a part of the legacyscheme, but might be worth noting?

Reply:
Yes, the aggregator and apiextensions types should be covered, too. I'll make sure this is included on the checklist.